devito/ir/clusters/algorithms.py

```diff
@@ -103,11 +108,20 @@ def guard(clusters):
                 # variables when the conditional is false
                 continue
 
-            guards = [(i.parent, conditions[i]) for i in dims]
-            guards.extend([(i.parent, negated[i]) for i in ndims])
-            cluster = PartialCluster(exprs, c.ispace, c.dspace, c.atomics, dict(guards))
+            if len(dims) > 0:
```

Comment:
The first case case `dims=()` correspond to the unconditional expressions that don't need guards.